### PR TITLE
Embroider fixes

### PR DESCRIPTION
**Pull request by anehx**
This PR contains various fixes needed for embroider support - however, there are various other issues in dependencies that need to be resolved before we can fully support embroider. Pipeline status can be seen in #1873

### CODE REVIEW

File: packages/testing/addon/mirage-graphql/deserialize.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default function deserialize(serialized) {
+  let decodedId = serialized.id;
+
+  try {
+    decodedId = atob(serialized.id).split(":")[1] || serialized.id;
+  } catch (e) {
+    // this is expected most times
```
Comment (luytena) [inline packages/testing/addon/mirage-graphql/deserialize.js:7]: Why?

Comment (luytena) [reply packages/testing/addon/mirage-graphql/deserialize.js:7]: Add comment

**Review by luytena — CHANGES_REQUESTED** [review]

File: packages/testing/addon/mirage-graphql/deserialize.js (hunk 1 above)
Comment (anehx) [reply packages/testing/addon/mirage-graphql/deserialize.js]: Done

**Review by luytena — APPROVED** [review]

**Pull request merged**